### PR TITLE
Refactor message parsing

### PR DESCRIPTION
**Pull request by erichaagdev**
This PR aims to improve the way data is passed to the command objects by using a single payload. The payload contains an object implementing a new `CommandMetadata` interface, which should be implemented by any callers to the bot (currently Discord and Rest).

### CODE REVIEW

File: src/main/java/com/gorlah/kappabot/command/CommandPayloadBuilder.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    private String sanitizeMessageToAdd(String message) {
+        message = removeLeadingQuoteIfNecessary(message);
+        message = removeTrailingQuoteIfNecessary(message);
+        message = replaceEscapedQuoteWithRegularQuote(message);
+
+        return message;
+    }
+
+    private String removeLeadingQuoteIfNecessary(String message) {
+        return !message.isEmpty() && message.charAt(0) == '\"' ? message.substring(1) : message;
```
Comment (ploober) [inline src/main/java/com/gorlah/kappabot/command/CommandPayloadBuilder.java]: `return message.startsWith("\"") ? ...`

File: src/main/java/com/gorlah/kappabot/command/CommandPayloadBuilder.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    private String removeTrailingQuoteIfNecessary(String message) {
+        return !message.isEmpty() && message.charAt(message.length() - 1) == '\"'
```
Comment (ploober) [inline src/main/java/com/gorlah/kappabot/command/CommandPayloadBuilder.java]: `return message.endsWith("\"") ? ...`

File: src/main/java/com/gorlah/kappabot/command/CommandPayloadBuilder.java (hunk 1 above)
Comment (erichaagdev) [reply src/main/java/com/gorlah/kappabot/command/CommandPayloadBuilder.java]: Oh that's much cleaner.  Nice!

Comment (erichaagdev) [reply src/main/java/com/gorlah/kappabot/command/CommandPayloadBuilder.java]: @ploober Fixed in latest commit.

**Review by ploober — APPROVED** [review]

**Pull request merged**